### PR TITLE
Fix SourceCodeKind setting

### PR DESCRIPTION
**Pull request by tmat**
For documents that have ParseOptions the SourceCodeKind should be set on the options and also on DocumentAttributes, for other documents it should be set on just the attributes.

### CODE REVIEW

File: src/Workspaces/Core/Portable/Workspace/Solution/DocumentState.cs [hunk 1]
```diff
@@ -29,10 +29,10 @@ internal partial class DocumentState : TextDocumentState
 
     // properties inherited from the containing project:
     public LanguageServices LanguageServices { get; }
-    private readonly ParseOptions? _options;
+    public ParseOptions? ParseOptions { get; }
```
Comment (tmat) [inline src/Workspaces/Core/Portable/Workspace/Solution/DocumentState.cs:32]: Simplify.

**Comment by tmat** [comment]
@dotnet/roslyn-ide ptal

File: src/Workspaces/Core/Portable/Workspace/Solution/DocumentState.cs [hunk 2]
```diff
@@ -397,12 +377,17 @@ private DocumentState SetParseOptions(ParseOptions options, bool onlyPreprocesso
...
             return this;
         }
 
-        return this.SetParseOptions(this.ParseOptions.WithKind(kind), onlyPreprocessorDirectiveChange: false);
+        if (ParseOptions != null)
+        {
+            return UpdateParseOptionsAndSourceCodeKind(ParseOptions.WithKind(kind), onlyPreprocessorDirectiveChange: false);
+        }
+
+        return UpdateAttributes(Attributes.With(sourceCodeKind: kind));
```
Comment (ToddGrun) [inline src/Workspaces/Core/Portable/Workspace/Solution/DocumentState.cs:390]: >return UpdateAttributes(Attributes.With(sourceCodeKind: kind)); [](http://example.com/codeflow?start=8&length=63)

This doesn't need to happen if the ParseOptions are set?

Comment (ToddGrun) [reply src/Workspaces/Core/Portable/Workspace/Solution/DocumentState.cs:390]: Oh, I see. UpdateParseOptionsAndSourceCodeKind does it when it creates the new DocumentState.

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**